src/diagram.js

```diff
@@ -1876,7 +1876,10 @@ dc_graph.diagram = function (parent, chartGroup) {
                 return name ? 'url(#' + arrow_id + ')' : null;
             })
             .each(function(e) {
-                d3.selectAll('#' + _diagram.arrowId(e, 'head') + ',#' + _diagram.arrowId(e, 'tail'))
+                var fillEdgeStroke = _diagram.edgeStroke.eval(e);
+                d3.selectAll('#' + _diagram.arrowId(e, 'head'))
+                    .attr('fill', _diagram.edgeStroke.eval(e));
+                d3.selectAll('#' + _diagram.arrowId(e, 'tail'))
```

Comment:
This is really obscure stuff! How on earth did you track down this performance problem?

Reply:
Yeah, this is really something I didn't know before. But in the profiler, apparently the original d3.selectAll take much more time than it should. Carlos suggests maybe the browser don't have optimization for the selector to do multiple selection while there may be some internal stuff to boost single selection. So the only thing I can test is using two d3.selectAll; and it turned out this makes a huge difference, at least on Chrome.

Reply:
Very interesting - I ought to profile dc.graph more. 

This area is already a bottleneck because of the nested `.each`, `.selectAll` where normally you could just nest selectAlls. So it's massively bad luck that I also chose the slowest selector inside the loop.